File: schemaorggo/government_permit.go
package schemaorggo

import "encoding/json"

// GovernmentPermit see : https://schema.org/GovernmentPermit
type GovernmentPermit struct {
	Permit

	typeContext

	// IssuedBy see : https://schema.org/issuedBy
	// The organization issuing the ticket or permit.
	// types : Organization
	IssuedBy []*Organization `json:"issuedBy,omitempty"`

	// IssuedThrough see : https://schema.org/issuedThrough
	// The service through with the permit was granted.
	// types : Service
	IssuedThrough []*Service `json:"issuedThrough,omitempty"`

	// PermitAudience see : https://schema.org/permitAudience
	// The target audience for this permit.
	// types : Audience
	PermitAudience []*Audience `json:"permitAudience,omitempty"`

	// ValidFor see : https://schema.org/validFor
	// The time validity of the permit.
	// types : Duration
	ValidFor []*Duration `json:"validFor,omitempty"`

	// ValidFrom see : https://schema.org/validFrom
	// The date when the item becomes valid.
	// types : DateTime
	ValidFrom []DateTime `json:"validFrom,omitempty"`

	// ValidIn see : https://schema.org/validIn
	// The geographic area where the permit is valid.
	// types : AdministrativeArea
	ValidIn []*AdministrativeArea `json:"validIn,omitempty"`

	// ValidUntil see : https://schema.org/validUntil
	// The date when the item is no longer valid.
	// types : Date
	ValidUntil []Date `json:"validUntil,omitempty"`
}

func (v GovernmentPermit) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.Permit.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.IssuedBy
		if len(v.IssuedBy) == 1 {
			value = v.IssuedBy[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["issuedBy"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.IssuedThrough
		if len(v.IssuedThrough) == 1 {
			value = v.IssuedThrough[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["issuedThrough"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.PermitAudience
		if len(v.PermitAudience) == 1 {
			value = v.PermitAudience[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["permitAudience"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.ValidFor
		if len(v.ValidFor) == 1 {
			value = v.ValidFor[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["validFor"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.ValidFrom
		if len(v.ValidFrom) == 1 {
			value = v.ValidFrom[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["validFrom"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.ValidIn
		if len(v.ValidIn) == 1 {
			value = v.ValidIn[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["validIn"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.ValidUntil
		if len(v.ValidUntil) == 1 {
			value = v.ValidUntil[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["validUntil"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v GovernmentPermit) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "GovernmentPermit"

	return data, nil
}

func (v GovernmentPermit) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/government_permit.php
<?php

class GovernmentPermit extends Permit implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'GovernmentPermit';
	
	/**
	 * The organization issuing the ticket or permit.
	 * see : https://schema.org/issuedBy
	 * @var \Organization|\Organization[]
	 */
	public var $issued_by;
	
	/**
	 * The service through with the permit was granted.
	 * see : https://schema.org/issuedThrough
	 * @var \Service|\Service[]
	 */
	public var $issued_through;
	
	/**
	 * The target audience for this permit.
	 * see : https://schema.org/permitAudience
	 * @var \Audience|\Audience[]
	 */
	public var $permit_audience;
	
	/**
	 * The time validity of the permit.
	 * see : https://schema.org/validFor
	 * @var \Duration|\Duration[]
	 */
	public var $valid_for;
	
	/**
	 * The date when the item becomes valid.
	 * see : https://schema.org/validFrom
	 * @var string|string[]
	 */
	public var $valid_from;
	
	/**
	 * The geographic area where the permit is valid.
	 * see : https://schema.org/validIn
	 * @var \AdministrativeArea|\AdministrativeArea[]
	 */
	public var $valid_in;
	
	/**
	 * The date when the item is no longer valid.
	 * see : https://schema.org/validUntil
	 * @var string|string[]
	 */
	public var $valid_until;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'GovernmentPermit'
		);
		
		$serialized = so_json_serialize( $this->issued_by );
		if ( ! empty( $serialized ) ) {
			$out['issuedBy'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->issued_through );
		if ( ! empty( $serialized ) ) {
			$out['issuedThrough'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->permit_audience );
		if ( ! empty( $serialized ) ) {
			$out['permitAudience'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->valid_for );
		if ( ! empty( $serialized ) ) {
			$out['validFor'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->valid_from );
		if ( ! empty( $serialized ) ) {
			$out['validFrom'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->valid_in );
		if ( ! empty( $serialized ) ) {
			$out['validIn'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->valid_until );
		if ( ! empty( $serialized ) ) {
			$out['validUntil'] = $serialized;
		}
		
		return $out;
	}
}
